packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'inner-outer', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]', function() {
-  expect(2);
+  ['@test Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]'](assert) {
+    assert.expect(1);
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
 
-  expectDeprecation(() => {
-    boot(() => {
-      appInstance.register('controller:application', Controller.extend({
-        'text': 'outer'
-      }));
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
+    }));
 
-      appInstance.register('component:my-component', Component.extend({
+    expectDeprecation(() => {
+      this.application.register('component:my-component', Component.extend({
         text: 'inner',
-        defaultLayout: compile('{{text}}-{{yield}}')
+        defaultLayout: this.compile('{{text}}-{{yield}}')
       }));
-    });
-  }, /Specifying `defaultLayout` to .+ is deprecated\./);
+    }, /Specifying `defaultLayout` to .+ is deprecated\./);
 
-  equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'inner-outer', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Assigning defaultLayout to a component should set it up as a layout if layout was found [DEPRECATED]', function() {
-  expect(2);
+  ['@test Assigning defaultLayout to a component should set it up as a layout if layout was found [DEPRECATED]'](assert) {
+    assert.expect(1);
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('components/my-component', compile('{{text}}-{{yield}}'));
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
 
-  expectDeprecation(() => {
-    boot(() => {
-      appInstance.register('controller:application', Controller.extend({
-        'text': 'outer'
-      }));
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
+    }));
 
-      appInstance.register('component:my-component', Component.extend({
-        text: 'inner',
-        defaultLayout: compile('should not see this!')
-      }));
+    expectDeprecation(() => {
+      this.addComponent('my-component', {
+        ComponentClass: Component.extend({
+          text: 'inner',
+          defaultLayout: this.compile('should not see this!')
+        }),
+        template: '{{text}}-{{yield}}'
+      });
+    }, /Specifying `defaultLayout` to .+ is deprecated\./);
```

Comment:
```js
expectDeprecation(() => {
      this.application.register('component:my-component', Component.extend({
        text: 'inner',
        defaultLayout: this.compile('{{text}}-{{yield}}')
      }));
    }, /Specifying `defaultLayout` to .+ is deprecated\./);
```
The above code fails here. Moving the expectDeprecation to `this.visit` works.

```js
    expectDeprecation(() => {
      this.visit('/').then(() => {
        let text = this.$('#wrapper').text().trim();
        assert.equal(text, 'inner-outer', 'The component is composed correctly');
      });
    }, /Specifying `defaultLayout` to .+ is deprecated\./);
```

I can see, we are calling applicationInstance.visit here. From there, we have called boot(). But i would like to know, when does the component registration takes place or from where?